File: tests/regress_simple.rs
#[test]
fn regress_simple() {
    let mut regress = liba::regress_simple::new(0.0, 0.0);
    let x = [0.0, 2.0, 4.0, 6.0, 8.0];
    let y = [1.0, 2.0, 3.0, 4.0, 5.0];
    let x_mean = liba::real_mean(&x);
    let y_mean = liba::real_mean(&y);
    regress.ols_(&x, &y, x_mean, y_mean);
    std::println!("y={}x+{}", regress.coef, regress.bias);
    for i in 0..x.len() {
        std::println!("{},{}", regress.evar(y[i]), regress.eval(x[i]));
    }
    regress.olsx(&x, &y, x_mean);
    std::println!("y={}x+{}", regress.coef, regress.bias);
    for i in 0..x.len() {
        std::println!("{},{}", regress.evar(y[i]), regress.eval(x[i]));
    }
    regress.olsy(&x, &y, y_mean);
    std::println!("y={}x+{}", regress.coef, regress.bias);
    for i in 0..x.len() {
        std::println!("{},{}", regress.evar(y[i]), regress.eval(x[i]));
    }
    regress.ols(&x, &y);
    std::println!("y={}x+{}", regress.coef, regress.bias);
    for i in 0..x.len() {
        std::println!("{},{}", regress.evar(y[i]), regress.eval(x[i]));
    }
    regress.zero();
}

File: quickjs/test/regress_simple.js
import * as a from "liba.so";
function assert(expr) {
    if (!eval(expr)) {
        throw Error(expr);
    }
}
var x = [0, 1, 2, 3, 4];
var y = [1, 2, 3, 4, 5];
var ctx = new a.regress_simple();
ctx.ols_(x, y, 2, 3);
console.log(ctx.evar(y[0]), ctx.eval(x[0]));
ctx.zero();
ctx.olsx(x, y, 2);
console.log(ctx.evar(y[0]), ctx.eval(x[0]));
ctx.zero();
ctx.olsy(x, y, 3);
console.log(ctx.evar(y[0]), ctx.eval(x[0]));
ctx.zero();
ctx.ols(x, y);
console.log(ctx.evar(y[0]), ctx.eval(x[0]));
ctx.zero();
assert("ctx.coef != undefined");
assert("ctx.bias != undefined");
ctx.coef = 1;
ctx.bias = 1;
